### PR TITLE
Streamline pypi release

### PR DESCRIPTION
**Pull request by Rowlando13**
Move to pyproject.toml. Tested with Test PyPi. Fixed issues with dependencies that were left out.

### CODE REVIEW

**Comment by Rowlando13** [comment]
Finally ready. https://test.pypi.org/project/mappymatch/0.2.2.post1/

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+  "pre-commit",
+  "pytest",
+  "sphinx==4.5.0",
+  "sphinx_rtd_theme==1.0.0",
+  "sphinxemoji==0.2.0",
+  ]
+release = [
+  "build",
+  "twine"
+]
```
Comment (nreinicke) [inline pyproject.toml:77]: How would you feel about consolidating the release block into the dev block for simplicity? And, how would you feel about just referring to the tests and docs dependencies blocks? I've tested the below locally and it seems to be working.

```suggestion
dev = [
  "mappymatch[tests]",
  "mappymatch[docs]",
  "coverage",
  "interrogate",
  "pre-commit",
  "build",
  "twine",
]
```

**Review by nreinicke — COMMENTED** [review]
This is great! I love the move to the `pyproject.toml` file. I left one small suggestion, curious to hear your thoughts. Thanks for doing this!

**Comment by Rowlando13** [comment]
Sounds good!

**Review by nreinicke — APPROVED** [review]

**Comment by nreinicke** [comment]
@Rowlando13 - can you share you PyPI account name with me so I can add you as a collaborator?

**Comment by nreinicke** [comment]
@Rowlando13 - perfect, just added you as a maintainer on pypi.org

**Comment by nreinicke** [comment]
@Rowlando13 - would you also be able to add me to the https://test.pypi.org/project/mappymatch/ project? (user is nreinicke)

**Comment by Rowlando13** [comment]
@nreinicke Done!

**Pull request merged**